app/src/main/java/com/bashkevich/androidfundamentals/model/MoviesRepository.kt

```diff
@@ -1,29 +1,54 @@
 package com.bashkevich.androidfundamentals.model
 
-import com.bashkevich.androidfundamentals.model.entity.Movie
-
-class MoviesRepository {
+import android.content.Context
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Transformations
+import androidx.lifecycle.map
+import com.bashkevich.androidfundamentals.model.entity.MovieActorCrossRef
+import com.bashkevich.androidfundamentals.model.viewobject.Movie
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+class MoviesRepository(applicationContext: Context) {
+    val database: MoviesDatabase = MoviesDatabase.create(applicationContext)
+
+    fun getAllMovies() = Transformations.map(database.moviesDao.getAllMovies()) { entities ->
+        entities.map { EntityMapper.fromMovieEntity(it) }
+    }
 
-    suspend fun getTopRatedMovies() = RetrofitModule.moviesApi.getTopRatedMovies()
-    suspend fun getNowPlayingMovies() = RetrofitModule.moviesApi.getNowPlayingMovies()
-    suspend fun getUpcomingMovies() = RetrofitModule.moviesApi.getUpcomingMovies()
-    suspend fun getPopularMovies() = RetrofitModule.moviesApi.getPopularMovies()
-    suspend fun getAllMovies(moviesIds: List<Int>) = moviesIds.distinct().map {
-        EntityMapper.convertToMovieFromDto(RetrofitModule.moviesApi.getMovieById(it))
+    suspend fun findMovieById(movieId: Int) = withContext(Dispatchers.IO) {
+        EntityMapper.fromMovieWithActors(database.moviesDao.getMovieWithActorsById(movieId))
```

Comment:
Just speaking loud - this will crash if no such movie, correct? Or it is handled in UI layer.

Reply:
there will be all movies from database, and we can't get an error as I understand